### PR TITLE
Feature/ms/t3.3 testing

### LINKED ISSUE #41: T11.2: Write tests for the components

### LINKED ISSUE #42: T9.6: Write tests for the components

### LINKED ISSUE #43: T6.4: Write tests for the components

### LINKED ISSUE #44: T3.3: Write tests for the components

### PR DESCRIPTION
**Pull request by msalze**
Closes #44 
Closes #43 
Closes #42 
Closes #41

### CODE REVIEW

File: src/main/java/com/agileavengers/icuconnectbackend/config/JwtRequestFilter.java [hunk 1]
```diff
@@ -31,17 +30,16 @@ public class JwtRequestFilter extends OncePerRequestFilter {
 
     @Override
     protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
-            FilterChain chain) throws ServletException, IOException {
+                                    FilterChain chain) throws ServletException, IOException {
 
-//        final String requestTokenHeader = request.getHeader("Authorization");
+        // final String requestTokenHeader = request.getHeader("Authorization");
```
Comment (solodezaldivar) [inline src/main/java/com/agileavengers/icuconnectbackend/config/JwtRequestFilter.java:35]: We can remove this

File: src/main/java/com/agileavengers/icuconnectbackend/service/ICommunityService.java [hunk 2]
```diff
@@ -7,12 +7,19 @@
...
     CommunityDto setupExampleCommunity();
+
     CommunityDto createCommunity(CommunityDto communityDto);
+
     Page<CommunityDto> getCommunities(int page, int size);
+
     CommunityDto getCommunity(Long id);
-//    Page<ReviewDto> getCommunityReviews(Long id, int page, int size);
+
+    //    Page<ReviewDto> getCommunityReviews(Long id, int page, int size);
```
Comment (solodezaldivar) [inline src/main/java/com/agileavengers/icuconnectbackend/service/ICommunityService.java:17]: Can we remove this?

File: src/main/java/com/agileavengers/icuconnectbackend/service/implementation/CommunityService.java [hunk 3]
```diff
@@ -116,17 +108,15 @@ public CommunityDto getCommunity(Long id) {
...
+        throw new ResponseStatusException(HttpStatus.NOT_FOUND, "entity not found");
     }
 
-//    @Override
-//    public Page<ReviewDto> getCommunityReviews(Long id, int page, int size) {
-//        Pageable pageable = PageRequest.of(page, size);
-//        Page<Review> reviewPage = reviewRepository.findAllByCommunity_Id(id, pageable);
-//        return reviewPage.map(reviewMapper::toDto);
-//    }
+    //    @Override
```
Comment (solodezaldivar) [inline src/main/java/com/agileavengers/icuconnectbackend/service/implementation/CommunityService.java:114]: Maybe this too? Or add a TODO so we don't forget?

**Review by solodezaldivar — CHANGES_REQUESTED** [review]
👍

File: src/main/java/com/agileavengers/icuconnectbackend/service/ICommunityService.java (hunk 2 above)
Comment (msalze) [reply src/main/java/com/agileavengers/icuconnectbackend/service/ICommunityService.java:17]: You are right. I removed it.

File: src/main/java/com/agileavengers/icuconnectbackend/service/implementation/CommunityService.java (hunk 3 above)
Comment (msalze) [reply src/main/java/com/agileavengers/icuconnectbackend/service/implementation/CommunityService.java:114]: You are right. I removed it.

File: src/main/java/com/agileavengers/icuconnectbackend/config/JwtRequestFilter.java (hunk 1 above)
Comment (msalze) [reply src/main/java/com/agileavengers/icuconnectbackend/config/JwtRequestFilter.java]: You are right. I removed it.

**Review by solodezaldivar — APPROVED** [review]

**Pull request merged**